google/datalab/notebook/static/codemirror/mode/bigquery.js

```diff
@@ -13,290 +13,109 @@
+      'FROM_BASE64', 'GENERATE_ARRAY', 'GENERATE_DATE_ARRAY', 'GREATEST', 'IEEE_DIVIDE', 'IS_INF',
+      'IS_NAN', 'JSON_EXTRACT', 'JSON_EXTRACT_SCALAR', 'LAG', 'LAST_VALUE', 'LEAD', 'LEAST',
+      'LENGTH', 'LN', 'LOG', 'LOG10', 'LOGICAL_AND', 'LOGICAL_OR', 'LOWER', 'LPAD', 'LTRIM', 'MAX',
+      'MD5', 'MIN', 'MOD', 'NET.HOST', 'NET.IPV4_FROM_INT64', 'NET.IPV4_TO_INT64',
+      'NET.IP_FROM_STRING', 'NET.IP_NET_MASK', 'NET.IP_TO_STRING', 'NET.IP_TRUNC',
+      'NET.PUBLIC_SUFFIX', 'NET.REG_DOMAIN', 'NET.SAFE_IP_FROM_STRING', 'NTH_VALUE', 'NTILE',
+      'OFFSET', 'ORDINAL', 'PARSE_DATE', 'PARSE_DATETIME', 'PARSE_TIME', 'PARSE_TIMESTAMP',
+      'PERCENT_RANK', 'POW', 'POWER', 'RAND', 'RANK', 'REGEXP_CONTAINS', 'REGEXP_EXTRACT',
+      'REGEXP_EXTRACT_ALL', 'REGEXP_REPLACE', 'REPEAT', 'REPLACE', 'REVERSE', 'ROUND',
+      'ROW_NUMBER', 'RPAD', 'RTRIM', 'SAFE_CONVERT_BYTES_TO_STRING', 'SAFE_DIVIDE', 'SAFE_OFFSET',
+      'SAFE_ORDINAL', 'SESSION_USER', 'SHA1', 'SHA256', 'SHA512', 'SIGN', 'SIN', 'SINH', 'SPLIT',
+      'SQRT', 'STARTS_WITH', 'STDDEV', 'STDDEV_POP', 'STDDEV_SAMP', 'STRING_AGG',
+      'STRPOS', 'SUBSTR', 'SUM', 'TAN', 'TANH', 'TIMESTAMP', 'TIMESTAMP_ADD', 'TIMESTAMP_DIFF',
+      'TIMESTAMP_MICROS', 'TIMESTAMP_MILLIS', 'TIMESTAMP_SECONDS', 'TIMESTAMP_SUB',
+      'TIMESTAMP_TRUNC', 'TIME_ADD', 'TIME_DIFF', 'TIME_SUB', 'TIME_TRUNC', 'TO_BASE64',
+      'TO_CODE_POINTS', 'TRIM', 'TRUNC', 'UNIX_DATE', 'UNIX_MICROS', 'UNIX_MILLIS', 'UNIX_SECONDS',
+      'UPPER', 'VARIANCE', 'VAR_POP', 'VAR_SAMP'
+    ]);
 
-      // DataLab extensions
-      'source',
-      'datestring'
+    var types = wordRegexp([
+      'INT64', 'FLOAT64', 'BOOL', 'STRING', 'BYTES', 'DATE', 'TIME', 'TIMESTAMP', 'ARRAY', 'STRUCT'
     ]);
 
-    var types = wordRegexp([]);
+    var operatorChars = /[.~^!<>*+=&|\\-]/;
+    var operatorWords = wordRegexp([
+      'NOT', 'LIKE', 'IS', 'AND', 'OR'
+    ]);
 
-    var operatorChars = /[*+\-<>=&|]/;
     var punctuationChars = /[{}\(\),;\[\]]/;
 
     function tokenBase(stream, state) {
       var ch = stream.next();
       curPunc = null;
-      if (stream.column() == 0 && ch == '#') {
+      if (ch == '#') {
         stream.skipToEnd();
         return 'comment';
+      } else if (ch === '-') {
+        var ch2 = stream.next();
+        if (ch2 === '-') {
+          stream.skipToEnd();
+          return 'comment';
+        }
       } else if (ch === '$' || ch === '?') {
         stream.match(/^[\w\d]*/);
         return 'variable-2';
-      } else if (ch === '<' && !stream.match(/^[\s\u00a0=]/, false)) {
-        stream.match(/^[^\s\u00a0>]*>?/);
-        return 'atom';
       } else if (ch === '"' || ch === '\'') {
         state.tokenize = tokenLiteral(ch);
         return state.tokenize(stream, state);
       } else if (ch === '`') {
-        state.tokenize = tokenOpLiteral(ch);
-        return state.tokenize(stream, state);
+        stream.eatWhile(/[^`]/);
```

Comment:
Curious why this string handling is different than the if block above for double and single quotes? And how if needed, this handles escapes?